src/System.Net.Http/src/System/Net/Http/Unix/CurlCallbacks.cs

```diff
@@ -414,5 +428,76 @@ private static void CheckForCompletedTransfers(SafeCurlMultiHandle multiHandle,
                 }
             }
         }
+
+        private async static void PollFunction(SafeCurlMultiHandle multiHandle)
+        {
+            while (true)
+            {
+                if (multiHandle.CancellationToken.IsCancellationRequested)
+                {
+                    return;
+                }
+
+                var count = multiHandle.FdList.Count;
+                if (count == 0)
+                {
+                    await Task.Delay(100).ConfigureAwait(false);
```

Comment:
Additionally, if the FdList.Count is 0, can't we stop polling until it's greater than 0?

And this 100ms period... is that recommended somewhere by libcurl?  I'm curious how we arrived at that value.